obspy/io/sac/core.py

```diff
@@ -381,7 +381,8 @@ def _internal_read_sac(buf, headonly=False, debug_headers=False, fsize=True,
     :return: A ObsPy Stream object.
     """
     # read SAC file
-    sac = SACTrace.read(buf, headonly=headonly, ascii=False, checksize=fsize)
+    sac = SACTrace.read(buf, headonly=headonly, ascii=False, checksize=fsize,
+                        encoding=kwargs.get('encoding', 'ASCII'))
```

Comment:
If I understand you correctly, `**kwargs` is intended to carry any format-specific arguments to the specific reader, so `_internal_read_sac` should get the `encoding` keyword from the `kwargs` dictionary.